src/platform/input/game-pads.js

```diff
@@ -68,62 +109,651 @@ const MAPS = {
+        previousAxes.push(...this.pad.axes);
+    }
+
+    /**
+     * The values from analog axes present on the GamePad. Values are between -1 and 1.
+     *
+     * @type {number[]}
+     */
+    get axes() {
+        return this.map.axes ? this.map.axes.map(a => this.pad.axes[MAPS_INDEXES.axes[a]] || 0) : [];
+    }
+
+    /**
+     * The buttons present on the GamePad. Some buttons may be null.
+     *
+     * @type {GamePadButton[]}
+     */
+    get buttons() {
+        return this.map.buttons ? this.map.buttons.map(b => this._getButton(b)) : [];
+    }
+
+    /**
+     * Make the gamepad vibrate.
+     *
+     * @param {number} intensity - Intensity for the vibration in the range 0 to 1.
+     * @param {number} duration - Duration for the vibration in milliseconds.
+     * @param {object} [options] - Options for special vibration pattern.
+     * @param {string} [options.type] - Type of pattern. Available types are "dual-rumble" and "vibration". Defaults to "vibration".
+     * @param {number} [options.startDelay] - Delay before the pattern starts, in milliseconds. Defaults to 0.
+     * @param {number} [options.strongMagnitude] - Intensity for strong actuators in the range 0 to 1. Defaults to intensity.
+     * @param {number} [options.weakMagnitude] - Intensity for weak actuators in the range 0 to 1. Defaults to intensity.
+     * @returns {Promise<boolean>} Return a Promise resulting in true if the pulse was successfully completed.
+     */
+    async pulse(intensity, duration, options) {
+        const actuators = this.pad.vibrationActuator ? [this.pad.vibrationActuator] : this.pad.hapticActuators || [];
+
+        if (actuators.length) {
+            const startDelay = options?.startDelay ?? 0;
+            const strongMagnitude = options?.strongMagnitude ?? intensity;
+            const weakMagnitude = options?.weakMagnitude ?? intensity;
+
+            const results = await Promise.all(
+                actuators.map(async (actuator) => {
+                    if (!actuator) {
+                        return true;
+                    }
+
+                    if (actuator.playEffect) {
+                        return await actuator.playEffect(actuator.type, {
+                            duration,
+                            startDelay,
+                            strongMagnitude,
+                            weakMagnitude
+                        });
+                    } else if (actuator.pulse) {
+                        await sleep(startDelay);
+                        return await actuator.pulse(intensity, duration);
+                    }
+
+                    return false;
+                })
+            );
+
+            for (let i = 0, l = results.length; i < l; i++) {
+                const result = results[i];
+
+                if (result === false || typeof result === 'string' && result !== 'complete') {
+                    return false;
+                }
+            }
+
+            return !!results.length;
```

Comment:
Can't we say that the pulse was successful if at least one actuator vibrated?

Reply:
That's a more positive way a thinking, that's updated!